### PR TITLE
Feat/emote sounds

### PR DESCRIPTION
**Pull request by esnyder232**
Sounds are working.
The coolhole options isn't created yet. Currently i'm hard coding the options to true in coolhole-util.js.

### CODE REVIEW

**Comment by esnyder232** [comment]
There is something I did different here compared to prod.
The place i'm actually doing the emote sounds is a bit different than in coolhole prod.
   In prod, we are doing the sounds AFTER the div is appended to chat.
   In new hole, we are doing the sounds in the FORMATTING of the div (before it is appended to chat).
It works out all the same really, i was just trying to avoid more changes to cytube files.

Next on the list is the emote overlaps (/soooy, /hearts, /gunshotleft), and that logic must be implented AFTER the div is appended to chat. Im gonna create a "postChatMessage(...)" to do that.
So if we did want emoteSounds to be called AFTER the div is appended to chat, we can move it to "postChatMessage()" once its created.

File: www/js/coolhole-util.js [hunk 1]
```diff
@@ -19,18 +19,284 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
...
+SFX.stack.isEnabled = true;
+
+// SFX Library
+/**
+* Can build this out to be something configurable
+* in the channel settings.
+*/
+SFX.mod.sounds = {
+    eugh: {
+        src: "https://ark.augint.net/CH_Emote_SFX-eugh/CH_Emote_SFX-eugh.wav",
```
Comment (streeeggs) [inline www/js/coolhole-util.js:134]: concern: Do we need to migrate these to yours?

File: www/js/coolhole-util.js [hunk 2]
```diff
@@ -19,18 +19,284 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
...
+        emote: "/eugh"
+    },
+    gunshot: {
+        src: "https://dontcodethis.com/images/Shotgun_Blast.wav",
+        emote: "/maths",
+    },
+    boogie: {
+        src: "https://dontcodethis.com/images/Boogie warning shot.wav",
```
Comment (streeeggs) [inline www/js/coolhole-util.js:142]: nit: This isn't a valid URL. This _probably_ works but it should be encoded, yea?

File: www/js/coolhole-util.js [hunk 3]
```diff
@@ -19,18 +19,284 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
...
+        // Only play if item doesn't have custom condition..
+        if( !sfxLibItem.hasOwnProperty( 'condition' )
+        // OR it DOES have condition and it PASSES
+        || ( sfxLibItem.hasOwnProperty( 'condition' ) && sfxLibItem.condition() ) ) {
+            // Init Audio item
+            var audio = new Audio( sfxLibItem.src );
+    
+            // Set to WAV
+            audio.type = 'audio/wav';
+            console.log( audio );
```
Comment (streeeggs) [inline www/js/coolhole-util.js:286]: nit: remove console log

File: www/js/coolhole-util.js [hunk 4]
```diff
@@ -19,18 +19,284 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
...
+    
+            // If a volume is set and it's a number, set it.
+            if ( sfxLibItem.hasOwnProperty( 'volume' ) && typeof sfxLibItem.volume === 'number' )
+            audio.volume = sfxLibItem.volume;
+            // If a playbackRate is set and it's a number, set it.
+            if ( sfxLibItem.hasOwnProperty( 'playbackRate' ) && typeof sfxLibItem.playbackRate === 'number' )
+            audio.playbackRate = sfxLibItem.playbackRate;
```
Comment (streeeggs) [inline www/js/coolhole-util.js:293]: nit: indentation

**Review by streeeggs — APPROVED** [review]

**Pull request merged**